File: snippets/python/discover_batch_points.py
from qdrant_client import QdrantClient, models

client = QdrantClient(url="http://localhost:6333")

discover_queries = [
    models.DiscoverRequest(
        target=[0.2, 0.1, 0.9, 0.7],
        context=[
            models.ContextExamplePair(
                positive=100,
                negative=718,
            ),
            models.ContextExamplePair(
                positive=200,
                negative=300,
            ),
        ],
        limit=10,
    ),
    models.DiscoverRequest(
        target=[0.5, 0.3, 0.2, 0.3],
        context=[
            models.ContextExamplePair(
                positive=342,
                negative=213,
            ),
            models.ContextExamplePair(
                positive=100,
                negative=200,
            ),
        ],
        limit=5,
    ),
]

client.discover_batch("{collection_name}", discover_queries)

File: snippets/csharp/discover_batch_points.cs
using Qdrant.Client;
using Qdrant.Client.Grpc;

var client = new QdrantClient("localhost", 6334);

var discoverPoints = new List<DiscoverPoints>
{
    new DiscoverPoints
    {
        CollectionName = "{collection_name}",
        Target = new TargetVector
        {
            Single = new VectorExample { Vector = new float[] { 0.2f, 0.1f, 0.9f, 0.7f }, }
        },
        Context =
        {
            new ContextExamplePair()
            {
                Positive = new VectorExample { Id = 100 },
                Negative = new VectorExample { Id = 718 }
            },
            new ContextExamplePair()
            {
                Positive = new VectorExample { Id = 200 },
                Negative = new VectorExample { Id = 300 }
            }
        },
        Limit = 10
    },
    new DiscoverPoints
    {
        CollectionName = "{collection_name}",
        Target = new TargetVector
        {
            Single = new VectorExample { Vector = new float[] { 0.5f, 0.3f, 0.2f, 0.3f }, }
        },
        Context =
        {
            new ContextExamplePair()
            {
                Positive = new VectorExample { Id = 342 },
                Negative = new VectorExample { Id = 213 }
            },
            new ContextExamplePair()
            {
                Positive = new VectorExample { Id = 100 },
                Negative = new VectorExample { Id = 200 }
            }
        },
        Limit = 10
    }
};
await client.DiscoverBatchAsync("{collection_name}", discoverPoints);
